### PR TITLE
`invalidateQuery()`-helper

### PR DESCRIPTION
**Pull request by KATT**
- allows you to use `trpc.invalidateQuery([path, input])` in a type-safe way

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -518,12 +521,101 @@ export const appRouter = createRouter()
...
+
+> - Your procedure needs to accept a `cursor` input of `any` type
+> - For more details read the [react-query docs](https://react-query.tanstack.com/reference/useInfiniteQuery)
+> - Example here is using Prisma - see their docs on [cursor-based pagination](https://www.prisma.io/docs/concepts/components/prisma-client/pagination#cursor-based-pagination)
+
+<details><summary>Example procedure (dummy code)</summary>
+
+```tsx
+import * as trpc from '@trpc/server';
+import { Context } from './[trpc]';
```
Comment (cyrus-za) [inline README.md:538]: I don't like importing from a slugified page. Or any page for that matter. But I guess its not wrong for example. I'd just put Context in types.ts or whatever. Not wrong though.

File: packages/react/src/createReactQueryHooks.ts [hunk 2]
```diff
@@ -60,9 +62,8 @@ export function createReactQueryHooks<
       TOutput
     >,
   ): UseQueryResult<TOutput, TRPCClientError> {
-    const path = pathAndArgs[0];
-    const input = pathAndArgs[1];
-    const cacheKey = [path, input ?? null];
+    const [path, input] = pathAndArgs;
+    const cacheKey = [path, input ?? null, CACHE_KEY_QUERY];
```
Comment (cyrus-za) [inline packages/react/src/createReactQueryHooks.ts:66]: nice. mamespaced keys to make sure it doesnt clash with other usages of react query

**Review by cyrus-za — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (KATT) [reply README.md:538]: I don't love it but i liked it more nor do I like circular reference but I like it more than creating a third file haha

File: README.md [hunk 3]
```diff
@@ -45,9 +45,12 @@ tRPC is a framework for building strongly typed RPC APIs with TypeScript. Altern
   - [Router middlewares](#router-middlewares)
   - [Data transformers](#data-transformers)
   - [Authorization](#authorization)
-  - [Server-side rendering (SSR / SSG)](#server-side-rendering-ssr--ssg)
-    - [Using `ssr.prefetchOnServer()` (recommended)](#using-ssrprefetchonserver-recommended)
-    - [Invoking directly](#invoking-directly)
+  - [React-specific helpers (`@trpc/react`)](#react-specific-helpers-trpcreact)
+    - [`useInfiniteQuery()`](#useinfinitequery)
+    - [`invalidateQuery()`](#invalidatequery)
+    - [`ssr()`: Server-side rendering (SSR / SSG)](#ssr-server-side-rendering-ssr--ssg)
```
[the remark this answers is not in the document]
Comment (KATT) [reply README.md:51]: `ssr()` is from `@trpc/react`

**Pull request merged**